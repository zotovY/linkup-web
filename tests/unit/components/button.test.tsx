/// <reference types="@types/jest" />
import { render, screen } from "../../test-utils";
import '@testing-library/jest-dom';
import 'jest-styled-components'

import Button from "@/components/button";
import { cleanup, fireEvent } from "@testing-library/react";

describe("Testing button component", () => {

    afterEach(cleanup);

    const props = {
        text: "hello there"
    }

    test("should handle click", () => {
        let value = false;
        const onSelect = () => value = !value;

        // Arrange
        const { getByText } = render(<Button onClick={onSelect}>{ props.text }</Button>);
        const component = screen.getByTestId("button");
        expect(getByText(props.text)).not.toBeFalsy();

        // Act
        fireEvent.click(component);

        // Assert
        expect(value).toBe(true);
    });

    test("should renders correctly", () => {
        const { asFragment } = render(<Button />);
        expect(asFragment()).toMatchSnapshot();
    });
});
