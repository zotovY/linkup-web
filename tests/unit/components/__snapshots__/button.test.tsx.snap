// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button component should renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 18px;
  font-weight: 500;
  cursor: pointer;
  border-radius: 12.5px;
  outline: none;
  -webkit-transition: background-color 150ms ease;
  transition: background-color 150ms ease;
  padding: 12.5px 30px;
  color: #f7f7fc;
  background-color: #5f2eea;
  border: 2px solid #5f2eea;
}

.c0 .content {
  border-radius: 15px;
  width: calc(100% + 20px);
  text-align: center;
}

.c0:hover {
  background-color: #6534ea;
}

<div
    class="react-ripples button-ripple button-container button-primary"
    style="position: relative; display: inline-flex; overflow: hidden;"
  >
    <button
      class="c0"
      data-testid="button"
      role="primary"
    />
    <s
      style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
    />
  </div>
</DocumentFragment>
`;
